lib/pure/asyncdispatch.nim

```diff
@@ -1026,6 +1038,10 @@ when defined(windows) or defined(nimdoc):
               # poll()
               GC_ref(pcd.ovl)
     )
+    # We need to protect our callback environment value, so GC will not free it
+    # accidentally.
+    ol.data.cell = system.protect(rawEnv(ol.data.cb))
```

Comment:
I think you need to watch full code
[Lines 974-1038](https://github.com/nim-lang/Nim/blob/devel/lib/pure/asyncdispatch.nim#L974-L1038)
Because if rawEnv() returns nil, then some regression happens in compiler... Just because ol.data.cb is under control of this function.